### PR TITLE
Add files for CoinGeckoPricesClient

### PR DESCRIPTION
**Pull request by emanuelgaspar**

### CODE REVIEW

File: tests/Trakx.CoinGecko.ApiClient.Tests/Integration/CoinGeckoClientTests.cs [hunk 1]
```diff
@@ -146,6 +146,27 @@ public async Task GetMarketData_should_return_valid_data_when_passing_valid_id()
         }
     }
 
+    [Fact]
+    public async Task GetAllPricesForSymbols_gets_prices_for_expected_symbols()
```
Comment (emanuelgaspar) [inline tests/Trakx.CoinGecko.ApiClient.Tests/Integration/CoinGeckoClientTests.cs:150]: `GetAllPricesForSymbols` is a little too complex for a bunch of unit tests, so adding an integration test that covers all scenarios - ranked and unranked coins - is faster and more robust.

File: tests/Trakx.CoinGecko.ApiClient.Tests/Unit/CoinGeckoClientTests.cs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 using Microsoft.Extensions.Caching.Memory;
+using Trakx.Common.ApiClient.Extensions;
 using Trakx.Common.Testing.Mocks;
 
 namespace Trakx.CoinGecko.ApiClient.Tests.Unit;
```
Comment (emanuelgaspar) [inline tests/Trakx.CoinGecko.ApiClient.Tests/Unit/CoinGeckoClientTests.cs:5]: The diff here is messed up.
It's not a comparison at all.

The tests were moved to `CoinGeckoPricesClientTests`.
The code from the "internal" file was moved here.

**Comment by emanuelgaspar** [comment]
The codacy issues are caused by the partial classes; Codacy can't understand that the fields are used or initialized in another file.

**Review by TiagoFroisPereira — APPROVED** [review]

File: src/Trakx.CoinGecko.ApiClient/CoinGeckoClients/CoinGeckoClient.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    internal const string MainQuoteCurrency = Constants.Usd;
+
+    private static readonly TimeSpan DefaultCacheLifeSpan = TimeSpan.FromDays(1);
+    private static readonly ILogger Logger = LoggerProvider.Create<CoinGeckoClient>();
+
+    private readonly IMemoryCache _cache;
+    protected readonly string? _typeName;
+
+    public CoinGeckoClient(
+        IMemoryCache cache,
```
Comment (goneneren) [inline src/Trakx.CoinGecko.ApiClient/CoinGeckoClients/CoinGeckoClient.cs:19]: [suggestion] I would call the variable memoryCache, as we also have distributed, and now hybrid cache

**Review by goneneren — APPROVED** [review]

**Review by joaofbvieira — APPROVED** [review]

File: src/Trakx.CoinGecko.ApiClient/CoinGeckoClients/CoinGeckoClient.cs (hunk 3 above)
Comment (emanuelgaspar) [reply src/Trakx.CoinGecko.ApiClient/CoinGeckoClients/CoinGeckoClient.cs:19]: I will rename in another PR.

**Pull request merged**